### PR TITLE
Challenges

### LINKED ISSUE #46: Functionality to challenge a friend
As of now we only have a challenge button in the sidebar view that does not do anything. We want this button to create a challenge request and to also be able to accept/decline the request in the sidebar view.

### PR DESCRIPTION
**Pull request by AnnaSkantz**
Now you can send challenges with a selected playlist. It is possible to accept/decline the request, but accepting will only lead to a console.log at the moment. This will be connected to the game functionality. 

resolves #46

### CODE REVIEW

**Review by alevarn — APPROVED** [review]
LGTM! The only problem was an error message in the console when you log in as the user that has received a challenge.

File: music-quiz/src/models/Model.js [hunk 1]
```diff
@@ -110,12 +110,12 @@ class Model {
...
 
-  newPendingRequest(searchUserData, requestType) {
-    // We do not want more than one request to/from a user respectively.
+  newPendingRequest(searchUserData, requestType, playlist = false) {
+    // We do not want more than one friend request to/from a user respectively.
     if (
       this.pending.find(
         (p) =>
-          p.type === requestType &&
+          p.type === "friendRequest" &&
```
Comment (alevarn) [inline music-quiz/src/models/Model.js:118]: What if it is a challenge?

File: music-quiz/src/models/Model.js [hunk 2]
```diff
@@ -153,6 +163,10 @@ class Model {
       username: name,
     };
 
+    if (playlist) {
+      newPending["playlist"] = playlist.id;
```
Comment (alevarn) [inline music-quiz/src/models/Model.js:167]: I think this is the problem. Try 
`newPending["playlist"] = playlist`
instead. This worked for me.

**Comment by asta12** [comment]
Now the bug should be fixed. It works for me now.

**Pull request merged**